csharp/src/Ice/Connection.cs

```diff
@@ -2345,7 +2335,7 @@ private int ParseMessage(ref MessageInfo info)
                             {
                                 _asyncRequests.Remove(info.RequestId);
 
-                                info.OutAsync.ResponseFrame = new IncomingResponseFrame(_communicator,
+                                var response = new IncomingResponseFrame(_communicator,
```

Comment:
Like for requests I would delay the construction of the frame when actually needed, i.e.: in the Response implementation.